src/sqlfluff/rules/L006.py

```diff
@@ -154,6 +155,17 @@ def _eval(self, context: RuleContext) -> EvalResultType:
                         check_after = True
 
             if check_before:
+                if context.dialect.name in ["sparksql"]:
+                    # NB: Should not check for whitespaces inside of Interval Literals for SparkSQL dialect
+                    # https://spark.apache.org/docs/latest/sql-ref-literals.html#interval-literal
+                    possible_interval_segment = self._find_segment(
+                        idx, context.segment.segments, before=True, step=-2
+                    )
+                    if (
+                        possible_interval_segment is not None
+                        and possible_interval_segment.raw.upper() == "INTERVAL"
+                    ):
+                        continue
```

Comment:
I think this is brittle but the best could come up with exploring the rule for the first time.

Would be very open to suggestions on how to improve this.